rcp/org.testeditor.rcp4.views.tcltestrun/src/org/testeditor/rcp4/views/tcltestrun/TclLauncherUi.xtend

```diff
@@ -92,45 +104,96 @@ class TclLauncherUi implements Launcher {
+		if (xmlResults.length > 0) {
+			writeTestResultFile(projectName, resultFile, xmlResults)
 		} else {
-			val newFile = File.createTempFile(expectedFile.name, ".xml", parentFolder)
-			if (expectedFile.exists) {
-				try {
-					Files.move(expectedFile, newFile)
-				} catch (IOException e) {
-					logger.error('''error during storage of test run result expectedFile='«expectedFile.path»' ''', e)
-					writeErrorFile(elementId, newFile)
+			writeErrorFile(elementId, resultFile)
+		}
+		JUnitCore.importTestRunSession(resultFile)
+	}
+
+	def void writeTestResultFile(String projectName, File resultFile, File[] xmlResults) {
+		val docBuilder = DocumentBuilderFactory.newInstance().newDocumentBuilder
+		val resultDoc = docBuilder.newDocument
+		val testRun = resultDoc.createElement("testrun")
+		resultDoc.appendChild(testRun)
+		var testCount = 0;
+		var failureCount = 0;
+		var errorsCount = 0;
+		var ignoreCount = 0;
+		for (file : xmlResults) {
+			val suiteDoc = docBuilder.parse(file)
+			val noteList = suiteDoc.childNodes
+			for (var i = 0; i < noteList.length; i++) {
+				if (noteList.item(i).nodeName.equals("testsuite")) {
+					testRun.appendChild(resultDoc.importNode(noteList.item(i), true))
+					testCount = testCount +
+						Integer.parseInt(noteList.item(i).attributes.getNamedItem("tests").nodeValue)
+					failureCount = failureCount +
+						Integer.parseInt(noteList.item(i).attributes.getNamedItem("failures").nodeValue)
+					errorsCount = errorsCount +
+						Integer.parseInt(noteList.item(i).attributes.getNamedItem("errors").nodeValue)
+					ignoreCount = ignoreCount +
+						Integer.parseInt(noteList.item(i).attributes.getNamedItem("skipped").nodeValue)
 				}
-			} else {
-				writeErrorFile(elementId, newFile)
-			}
-			JUnitCore.importTestRunSession(newFile)
-			try {
-				java.nio.file.Files.delete(newFile.toPath)
-			} catch (IOException e) {
-				logger.warn("error during removal of obsolete test result newFile='{}'", newFile.path)
 			}
 		}
+		testRun.attributeNode = createAttribute("name", resultDoc, "java")
+		testRun.attributeNode = createAttribute("project", resultDoc, projectName)
+		testRun.attributeNode = createAttribute("tests", resultDoc, Integer.toString(testCount))
+		testRun.attributeNode = createAttribute("started", resultDoc, Integer.toString(testCount))
+		testRun.attributeNode = createAttribute("failures", resultDoc, Integer.toString(failureCount))
+		testRun.attributeNode = createAttribute("errors", resultDoc, Integer.toString(errorsCount))
+		testRun.attributeNode = createAttribute("ignored", resultDoc, Integer.toString(ignoreCount))
+		val transformerFactory = TransformerFactory.newInstance();
+		val transformer = transformerFactory.newTransformer();
+		val source = new DOMSource(resultDoc);
+		transformer.transform(source, new StreamResult(resultFile));
+	}
+
+	def Attr createAttribute(String attributeName, Document doc, String attributeValue) {
```

Comment:
If you move the `doc` parameter to the front you can write: `resultDoc.createAttribute("name", "java")` which is neat.